lib/Doctrine/Migrations/Migrator.php

```diff
@@ -179,6 +179,15 @@ private function executeMigration(
             foreach ($migrationsToExecute as $version) {
                 $versionExecutionResult = $version->execute($direction, $migratorConfiguration);
 
+                $toSchema = $versionExecutionResult->getToSchema();
+
+                // capture the to Schema for the migration so we have the ability to use
+                // it as the from Schema for the next migration when we are running a dry run
+                // $toSchema may be null in the case of skipped migrations
+                if ($toSchema !== null) {
```

Comment:
Can you use

``` php
if (! $versionExecutionResult->isSkipped()) {
        $toSchema = $versionExecutionResult->getToSchema();
        $migratorConfiguration->setFromSchema($toSchema);
}
```

here?

Reply:
I could, but then I think PHPStan will still complain because getToSchema() return type hint can be null. I could remove the optional return type hint and throw an exception when the getter is called and toSchema is null. This should only happen when isSkipped is true.

Reply:
This is fixed now.